ext/cl_khr_integer_dot_product.asciidoc

```diff
@@ -0,0 +1,212 @@
+// Copyright 2020-2021 The Khronos Group. This work is licensed under a
+// Creative Commons Attribution 4.0 International License; see
+// http://creativecommons.org/licenses/by/4.0/
+
+[[cl_khr_integer_dot_product]]
+== Integer dot product
+
+This extension adds support for SPIR-V instructions and OpenCL C built-in
+functions to compute the dot product of vectors of integers.
+
+=== General Information
+
+==== Name Strings
+
+`cl_khr_integer_dot_product`
+
+==== Version History
+
+[cols="1,1,3",options="header",]
+|====
+| *Date*     | *Version* | *Description*
+| 2021-06-17 | 1.0.0     | Initial version.
+|====
+
+==== Dependencies
+
+This extension is written against the OpenCL Specification Version 3.0.6,
+and OpenCL C Specification Version 3.0.6 and OpenCL Environment Specification
+Version 3.0.6.
+
+This extension requires OpenCL 1.0.
+
+==== Contributors
+
+Kévin Petit, Arm Ltd. +
+Jeremy Kemp, Imagination Technologies +
+Ben Ashbaugh, Intel +
+Ruihao Zhang, Qualcomm +
+Stuart Brady, Arm Ltd +
+Balaji Calidas, Qualcomm +
+Ayal Zaks, Intel +
+
+=== New API Enums
+
+Accepted value for the _param_name_ parameter to *clGetDeviceInfo*:
+
+[source,c]
+----
+
+CL_DEVICE_INTEGER_DOT_PRODUCT_INPUT_4x8BIT_PACKED_KHR      (1 << 0)
+CL_DEVICE_INTEGER_DOT_PRODUCT_INPUT_4x8BIT_KHR             (1 << 1)
+
+CL_DEVICE_INTEGER_DOT_PRODUCT_CAPABILITIES_KHR              0x1073
+----
+
+=== New OpenCL C Functions
+
+This extension defines a number of new functions that operate on vectors
+of integers. The exact function overloads available depend on the features
+supported by the target device.
+
+[source,c]
+----
+uint dot(uchar4 a, uchar4 b);
+int dot(char4 a, char4 b);
+int dot(uchar4 a, char4 b);
+int dot(char4 a, uchar4 b);
+
+uint dot_acc_sat(uchar4 a, uchar4 b, uint acc);
+int dot_acc_sat(char4 a, char4 b, int acc);
+int dot_acc_sat(uchar4 a, char4 b, int acc);
+int dot_acc_sat(char4 a, uchar4 b, int acc);
+
+uint dot_4x8packed_uu_uint(uint a, uint b);
+int dot_4x8packed_ss_int(uint a, uint b);
+int dot_4x8packed_us_int(uint a, uint b);
+int dot_4x8packed_su_int(uint a, uint b);
+
+uint dot_acc_sat_4x8packed_uu_uint(uint a, uint b, uint acc);
+int dot_acc_sat_4x8packed_ss_int(uint a, uint b, int acc);
+int dot_acc_sat_4x8packed_us_int(uint a, uint b, int acc);
+int dot_acc_sat_4x8packed_su_int(uint a, uint b, int acc);
+----
+
+=== Modifications to the OpenCL API Specification
+
+(Modify Section 4.2, *Querying Devices*) ::
++
+--
+
+(Add the following to Table 4.3, _Device Queries_) ::
++
+--
+
+[cols="2,2,4",options="header"]
+|====
+| cl_device_info
+| Return Type
+| Description
+
+| `CL_DEVICE_INTEGER_DOT_PRODUCT_CAPABILITIES_KHR`
+| `cl_device_integer_dot_product_capabilities_khr`
```

Comment:
Consider using the auto-generated asciidoctor macros `{CL_DEVICE_INTEGER_DOT_PRODUCT_CAPABILITIES_KHR}` and `{cl_device_integer_dot_product_capabilities_khr_type}` so these table cells wrap and do not widen the table excessively.

Reply:
I've tried this and it seems to have no impact on the output for me. Is the behaviour specific to given versions of asciidoctor?

Reply:
Are you looking at the HTML output or the PDF output?

The PDF output looks reasonably OK to me without the macros, though the table cells wrap arbitrarily rather than at underscore boundaries.  The HTML output is very wide without the macros, though.

Reply:
I was looking at the PDF only. I can see the impact on the HTML spec as well. I've added a commit to use the macros.